qa/tasks/quiescer.py

```diff
@@ -0,0 +1,236 @@
+            raise RuntimeError(f"Error parsing quiesce response: {e}")
+
+        self.logger.info(f"Successfully quiesced, set_id: {set_id}, quiesce duration: {duration}")
+        self.stopping.wait(duration)
+
+        # release the root
+        rc, stdout = self.tell_quiesce_leader(
+            "--set-id", set_id,
+            "--release",
+            "--await"
+        )
+        
+        if rc != 0:
+            rcinfo = f"{-rc} ({errno.errorcode.get(-rc, 'Unknown')})"
+            self.logger.error(f"Coulding release root with rc: {rcinfo}, stdout:\n{stdout}")
+            raise RuntimeError(f"Error releasing root: {rcinfo}")
+
+
+    def _run(self):
+        try:
+            self.stopping.wait(self.initial_delay)
+
+            while not self.stopping.is_set():
+                duration = self.next_quiesce_duration()
+                self.do_quiesce(duration)
+                # now we sleep to maintain the quiesce factor
+                self.stopping.wait((duration/self.quiesce_factor) - duration)
+
+        except Exception as e:
+            self.set_thrasher_exception(e)
+            self.logger.exception("exception:")
+            # allow successful completion so gevent doesn't see an exception...
+
+    def stop(self):
+        self.tell_quiesce_leader( "--cancel", "--all" )
+        self.stopping.set()
+
+
+def stop_all_quiescers(thrashers):
+    for thrasher in thrashers:
+        if not isinstance(thrasher, Quiescer):
+            continue
+        thrasher.stop()
+        thrasher.join()
+        if thrasher.exception is not None:
+            raise RuntimeError(f"error during scrub thrashing: {thrasher.exception}")
+
+
+@contextlib.contextmanager
+def task(ctx, config):
+    """
+    Stress test the mds by randomly quiescing the whole FS while another task/workunit
+    is running.
+    Example config:
+
+    - fwd_scrub:
+      scrub_timeout: 300
+      sleep_between_iterations: 1
+    """
+
+    mds_cluster = MDSCluster(ctx)
+
+    if config is None:
+        config = {}
+    assert isinstance(config, dict), \
+        'quiescer task only accepts a dict for configuration'
+    mdslist = list(teuthology.all_roles_of_type(ctx.cluster, 'mds'))
+    assert len(mdslist) > 0, \
+        'quiescer task requires at least 1 metadata server'
+
+    (first,) = ctx.cluster.only(f'mds.{mdslist[0]}').remotes.keys()
+    manager = ceph_manager.CephManager(
+        first, ctx=ctx, logger=log.getChild('ceph_manager'),
+    )
+
+    # make sure everyone is in active, standby, or standby-replay
+    log.info('Wait for all MDSs to reach steady state...')
+    status = mds_cluster.status()
+    while True:
+        steady = True
+        for info in status.get_all():
+            state = info['state']
+            if state not in ('up:active', 'up:standby', 'up:standby-replay'):
+                steady = False
+                break
+        if steady:
+            break
+        sleep(2)
+        status = mds_cluster.status()
```

Comment:
This check should be moved to the Quiescer and it should use `fs.wait_for_daemons()`.

Reply:
I believe you meant `moved to Thrasher`? I decided to wait with this refactor because it quickly grew in volume when I started doing it, so for now I'll leave this code duplication and make an enhancement ticket to address this later, if you agree

Reply:
> I believe you meant `moved to Thrasher`?

No, `Quiescer`.

>  I decided to wait with this refactor because it quickly grew in volume when I started doing it, so for now I'll leave this code duplication and make an enhancement ticket to address this later, if you agree

why? I think you can literally drop-in `fs.wait_for_daemons()` for this loop.